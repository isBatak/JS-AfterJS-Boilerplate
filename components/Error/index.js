import React from 'react';

export default class Error extends React.Component {

  render() {
    const {code, children} = this.props;

    return (
      <div>
        <h1>{code}</h1>
        <h4>{children}</h4>
      </div>
    );
  }
}
