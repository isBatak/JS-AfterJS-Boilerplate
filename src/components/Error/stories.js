import React from 'react';
import {storiesOf} from '@storybook/react';
import Error from './';

storiesOf('Error', module)
  .add('not found', () => (
    <Error code="404">Not found</Error>
  ))
  .add('forbidden', () => (
    <Error code="403">Forbidden</Error>
  ));
