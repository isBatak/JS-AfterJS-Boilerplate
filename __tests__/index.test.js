/* eslint-env jest */

import Enzyme, {shallow} from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Error from '../components/Error';

Enzyme.configure({adapter: new Adapter()});

describe('With Enzyme', () => {
  it('Error shows "Welcome to next.js!"', () => {
    const app = shallow(<Error code="Welcome to next.js!" />);

    expect(app.find('h1').text()).toEqual('Welcome to next.js!');
  });
});

describe('With Snapshot Testing', () => {
  it('Error shows "Welcome to next.js!"', () => {
    const component = renderer.create(<Error />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
