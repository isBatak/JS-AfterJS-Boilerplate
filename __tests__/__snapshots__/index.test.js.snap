// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Error shows "Welcome to next.js!" 1`] = `
<div>
  <h1 />
  <h4 />
</div>
`;
